core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1018 @@
+
+    expect(explorationId).toBe('id1');
+  });
+
+  it('should return the version of exploration', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let version = questionPlayerEngineService.getExplorationVersion();
+    expect(version).toBe(1);
+  });
+
+  it('should clear all questions', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['questions'].length).toBe(3);
+
+    questionPlayerEngineService.clearQuestions();
+
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['questions'].length).toBe(0);
+  });
+
+  it('should return the language code', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let languageCode = questionPlayerEngineService.getLanguageCode();
+
+    expect(languageCode).toBe('en');
```

Comment:
Again, need at least two possibilities. Otherwise for all we know this could be returning the site language code.

Reply:
done.